zkvm/src/contract.rs

```diff
@@ -127,6 +118,11 @@ impl Output {
         buf
     }
 
+    /// Constructs a new Output from payload and predicate
+    pub fn new(payload: Vec<FrozenItem>, predicate: Predicate) -> Self {
```

Comment:
We can allow user construct the struct (mark all Output fields `pub`) themselves because there's no invariant to protect between the predicate and items. So we don't need to have helper methods that imply otherwise and do not make it more convenient: actually, less, since method argument names are not visible at the callsite, but struct field names are.